rest/client.go

```diff
@@ -19,88 +18,98 @@ type TreeFilter struct {
-	GetProviderNodeIDsByType(ctx context.Context, provider, originType string) ([]uuid.UUID, error)
-	GetOriginNodeID(ctx context.Context, origin models.Origin) (uuid.UUID, error)
+	DuplicateNode(ctx context.Context, source uuid.UUID, destination uuid.UUID, suffix string) (models.DuplicateNodeResponse, error)
+
+	GetAncestors(ctx context.Context, id uuid.UUID, height int, nodeTypes ...string) (models.GetAncestorsResponse, error)
+	GetCompany(ctx context.Context, id uuid.UUID) (models.GetCompanyResponse, error)
+	GetSubtree(ctx context.Context, id uuid.UUID, filter TreeFilter, continuationToken string) (models.GetSubtreeResponse, error)
+	GetSubtreeCount(ctx context.Context, id uuid.UUID, nodeTypes ...string) (models.GetSubtreeCountResponse, error)
+
+	LockNode(ctx context.Context, id uuid.UUID, recursive bool) error
+	UnlockNode(ctx context.Context, id uuid.UUID, recursive bool) error
+
+	GetOrigins(ctx context.Context, provider, continuationToken string, limit int) (models.GetOriginsResponse, error)
+	GetOriginsByType(ctx context.Context, provider, originType, continuationToken string, limit int) (models.GetOriginsResponse, error)
+	GetProviderNodeIDs(ctx context.Context, provider, continuationToken string, limit int) (models.GetNodesByPartialOriginResponse, error)
+	GetProviderNodeIDsByType(ctx context.Context, provider, originType, continuationToken string, limit int) (models.GetNodesByPartialOriginResponse, error)
+	GetOriginNodeID(ctx context.Context, origin models.Origin) (models.GetNodeByOriginResponse, error)
 }
 
 type client struct {
 	*rest.Client
+	clientID string
 }
 
 func WithStage(stage string) rest.Option {
 	if stage == stages.StageProd {
-		return rest.WithBaseURL("https://api.hierarchy.enlight.skf.com")
+		return rest.WithBaseURL("https://api.hierarchy.enlight.skf.com/v2/")
 	}
 
-	return rest.WithBaseURL(fmt.Sprintf("https://api.%s.hierarchy.enlight.skf.com", stage))
+	return rest.WithBaseURL(fmt.Sprintf("https://api.%s.hierarchy.enlight.skf.com/v2/", stage))
 }
 
-func NewClient(opts ...rest.Option) HierarchyClient {
+func WithClientID(clientID string) rest.Option {
+	clientHeader := "X-Client-Id"
+	return rest.WithDefaultHeader(clientHeader, clientID)
+}
+
+func NewClient(clientID string, opts ...rest.Option) HierarchyClient {
 	restClient := rest.NewClient(
 		append([]rest.Option{
 			// Defaults to production stage if no option is supplied
 			WithStage(stages.StageProd),
 		}, opts...)...,
 	)
 
-	return &client{Client: restClient}
+	return &client{Client: restClient, clientID: clientID}
 }
 
-func (c *client) GetNode(ctx context.Context, id uuid.UUID) (models.Node, error) {
+func (c *client) GetNode(ctx context.Context, id uuid.UUID) (models.GetNodeResponse, error) {
 	request := rest.Get("nodes/{node}").
 		Assign("node", id).
 		SetHeader("Accept", "application/json")
 
-	var response models.WebmodelsNode
+	var response models.GetNodeResponse
 	if err := c.DoAndUnmarshal(ctx, request, &response); err != nil {
```

Comment:
How do we want to handle Problems with this client?

I think we might need to do some changes to the main client library to support it. Doing it like this will remove all information regarding the Problem and just focus on the HTTP status code (400, 404, etc)

Reply:
I guess we can do this later then?